src/main/java/parser/Parser.java

```diff
@@ -302,6 +331,25 @@ public Command processDeleteAuthorMangaCommand(String userInput) throws TantouEx
         throw new TantouException("Invalid delete command provided!");
     }
 
+    //@@author sarahchow03
+    /**
+     * Processes the user input to create a Sales command for either an Author.
+     *
+     * If the input is a valid sales command, it returns an `DeleteMangaCommand`.
+     *
+     * @param userInput the raw input string provided by the user,
+     *                  which should include the author, manga, quantity sold, and unit price.
+     * @return a Command object that adds the sales data for the specified manga.
+     * @throws TantouException if the user input is missing a parameter.
+     */
+    public Command processAddSalesCommand(String userInput) throws TantouException {
+        if (isValidSalesCommand(userInput)) {
+            return new AddSalesCommand(userInput);
+        }
+        throw new TantouException("Invalid sales command provided!"
+                + " You need to provide the author, manga, quantity sold, and unit price.");
```

Comment:
Good job making it clear to the user what they need to provide. Perhaps we can give even more information about the specific option flags needed (e.g. `-a`, `-m`) so that they can form a valid command?